zyte_common_items/items.py

```diff
@@ -252,3 +376,123 @@ class Product(_ProductBase):
+    #: ``currencyRaw``), e.g. ``"10500.99"``.
+    #:
+    #: If ``regularPrice`` is not ``None``, ``price`` should always be lower
+    #: than ``regularPrice``.
+    price: Optional[str] = None
+
+    #: Product identifier, unique within an e-commerce website.
+    #:
+    #: It may come in the form of an SKU or any other identifier, a hash, or
+    #: even a URL.
+    productId: Optional[str] = None
+
+    #: Price at which the product was being offered in the past, and which is
+    #: presented as a reference next to the current price.
+    #:
+    #: It may be labeled as the original price, the list price, or the maximum
+    #: retail price for which the product is sold.
+    #:
+    #: See ``price`` for format details.
+    #:
+    #: If ``regularPrice`` is not ``None``, it should always be higher than
+    #: ``price``.
+    regularPrice: Optional[str] = None
+
+    #: Main URL from which the product data could be extracted.
+    url: Optional[str] = None
+
+
+@attrs.define(slots=True, kw_only=True)
+class ProductList(Item):
+    """Product list from a product listing page of an e-commerce webpage.
+
+    It represents, for example, a single page from a category.
+
+    The :attr:`url` attribute is the only required attribute, all other fields
+    are optional.
+    """
+
+    #: Webpage `breadcrumb trail`_.
+    #:
+    #: .. _Breadcrumb trail: https://en.wikipedia.org/wiki/Breadcrumb_navigation
+    breadcrumbs: Optional[List[Breadcrumb]] = None
+
+    #: Canonical form of the URL, as indicated by the website.
+    #:
+    #: See also ``url``.
+    canonicalUrl: Optional[str] = None
+
+    #: Name of the product listing as it appears on the webpage
+    #: (no post-processing).
+    #:
+    #: For example, if the webpage is one of the pages of the Robots category,
+    #: ``categoryName`` is ``'Robots'``.
+    categoryName: Optional[str] = None
+
+    #: Data extraction process metadata.
+    metadata: Optional[Metadata] = None
+
+    #: Number of the current page.
+    #:
+    #: It should only be extracted if the webpage shows a page number.
+    #:
+    #: It must be 1-based. For example, if the first page of a listing is
+    #: numbered as 0 on the website, it should be extracted as `1` nonetheless.
+    pageNumber: Optional[int] = None
+
+    #: Link to the next page.
+    paginationNext: Optional[Link] = None
+
+    #: List of products.
+    #:
+    #: It only includes product information found in the product listing page
+    #: itself. Product information that requires visiting each product URL is
+    #: not meant to be covered.
+    #:
+    #: The order of the products reflects their position on the rendered page.
+    #: Product order is top-to-bottom, and left-to-right or right-to-left
+    #: depending on the webpage locale.
+    products: Optional[List[ProductListItem]] = None
+
+    #: Main URL from which the data has been extracted.
+    #:
+    #: See also ``canonicalUrl``.
+    url: str
```

Comment:
Should we add the `metadata` field as well?

Reply:
It is a few fields above (fields are in alphabetical order).

Reply:
My bad, I totally missed it!